includes/class-simple-local-avatars.php

```diff
@@ -678,4 +717,159 @@ public function get_avatar_rest( $user ) {
+				$sla->assign_new_user_avatar( (int) $avatar_id, $user->ID );
+
+				// Check that it worked.
+				$is_migrated = get_user_meta( $user->ID, 'simple_local_avatar', true );
+
+				if ( ! empty( $is_migrated ) ) {
+					// Build array of user IDs.
+					$migrations['wp_user_avatar'][] = $user->ID;
+
+					// Record the user IDs so we don't process a second time.
+					$is_saved = update_option( 'simple_local_avatars_migrations', $migrations );
+
+					// Record how many avatars we migrate to be used in our messaging.
+					if ( $is_saved ) {
+						$count++;
+					}
+				}
+			}
+		}
+
+		return $count;
+
+	}
+
+	/**
+	 * Migrate the user's avatar data away from WP User Avatar/ProfilePress via the dashboard.
+	 *
+	 * Sends the number of avatars processed back to the AJAX response before stopping execution.
+	 *
+	 * @return void
+	 */
+	public function ajax_migrate_from_wp_user_avatar() {
+		// Bail early if nonce is not available.
+		if ( empty( $_POST['migrateFromWpUserAvatarNonce'] ) ) {
+			die;
+		}
+
+		// Bail early if nonce is invalid.
+		if ( ! wp_verify_nonce( $_POST['migrateFromWpUserAvatarNonce'], 'migrate_from_wp_user_avatar_nonce' ) ) {
+			die();
+		}
+
+		// Run the migration script and store the number of avatars processed.
+		$count = $this->migrate_from_wp_user_avatar();
+
+		// Create the array we send back to javascript here.
+		$array_we_send_back = array( 'count' => $count );
+
+		// Make sure to json encode the output because that's what it is expecting.
+		echo wp_json_encode( $array_we_send_back );
+
+		// Make sure you die when finished doing ajax output.
+		wp_die();
+
+	}
+
+	/**
+	 * Migrate the user's avatar data from WP User Avatar/ProfilePress via the command line.
+	 *
+	 * ## OPTIONS
+	 *
+	 * [--yes]
+	 * : Skips the confirmations (for automated systems).
+	 *
+	 * ## EXAMPLES
+	 *
+	 *     $ wp simple-local-avatars migrate wp-user-avatar
+	 *     Success: Successfully migrated 5 avatars.
+	 *
+	 * @param array|null $args       The arguments.
+	 * @param array|null $assoc_args The associative arguments.
+	 *
+	 * @return void
+	 */
+	public function wp_cli_migrate_from_wp_user_avatar( $args = null, $assoc_args = null ) {
+
+		// Argument --yes to prevent confirmation (for automated systems).
+		if ( ! isset( $assoc_args['yes'] ) ) {
+			WP_CLI::confirm( esc_html__( 'Do you want to migrate avatars from WP User Avatar?', 'simple-local-avatars' ) );
+		}
+
+		// Run the migration script and store the number of avatars processed.
+		$count = $this->migrate_from_wp_user_avatar();
+
+		// Error out if we don't process any avatars.
+		if ( 0 === absint( $count ) ) {
+			WP_CLI::error( esc_html__( 'No avatars were migrated from WP User Avatar.', 'simple-local-avatars' ) );
+		}
+
+		WP_CLI::success(
+			sprintf(
+				'%s %s %s',
+				esc_html__( 'Successfully migrated', 'simple-local-avatars' ),
+				esc_html( $count ),
+				esc_html__( 'avatars from WP User Avatar.', 'simple-local-avatars' )
```

Comment:
Strings for translations should not be fragments - not all languages construct sentences the same way as English, translators will not have full context in GlotPress, and even in English you'll run into a plural problem if you migrate 1 avatar. You'll want to use `_n()` instead.

Reply:
I change the wording to a single string for the CLI version as well as the ajax version. Makes things easier on everyone.